File: models/radiology/RadiologyImage.js
// models/RadiologyImage.js
import mongoose from "mongoose";

const RadiologyImageSchema = new mongoose.Schema({
  hospitalId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Hospital",
    required: true
  },
  branchId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Branch",
    required: true
  },

  radiologyStudyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "RadiologyStudy",
    required: true
  },

  imageUrl: String,   // PACS URL
  dicomId: String

}, { timestamps: true });

export default mongoose.model("RadiologyImage", RadiologyImageSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/RadiologyImage.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "radiology_images")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RadiologyImage extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "radiology_study_id", nullable = false)
    private RadiologyStudy radiologyStudy;

    @Column(name = "image_url")
    private String imageUrl; // PACS URL

    @Column(name = "dicom_id")
    private String dicomId;
}
